### PR TITLE
refactor provisioning: Make checkRemoveConfigMapping more efficient

### PR DESCRIPTION
**Pull request by hansbogert**

### CODE REVIEW

File: api/provisioning.go [hunk 1]
```diff
@@ -1800,64 +1759,70 @@ func checkConfigMapping(applied []Configlet, newconfiglets []Configlet) (bool,
...
-		default:
-			return false, nil, nil, nil, nil,
-				errors.Errorf("Configlet [%s] Invalid Type [%s]", configlet.Name, configlet.Type)
-		}
+	configletKeys, configletNames, builderKeys, builderNames, err2 := splitToConfigletAndBuilder(configletsToRemain)
+	if err2 != nil {
+		return false, nil, nil, nil, nil, nil, nil,nil, nil, err2
 	}
 
 	if len(configletNames) == 0 && len(builderNames) == 0 {
```
Comment (hansbogert) [inline api/provisioning.go]: @cheynearista personally I don't get this special case of both being length 0, what was/is this supposed to solve? Maybe I can add it as a comment once I know the rationale

Comment (cheynearista) [reply api/provisioning.go]: @hansbogert I think this was to handle the situation where applied []Configlet matches rmConfiglets []Configlet (i.e. User is removing all configlets that are currently applied). In this case actionReqd would need to be true so the calling func can create the 'action' to removed them.

Comment (hansbogert) [reply api/provisioning.go]: Then my `actionreqd = true` on line https://github.com/aristanetworks/go-cvprac/pull/70/files#diff-af964366e44ed2bd45c6078a428dc200R1813 doesn't make much sense.
Would it by equivalent if we do a check for the following:

`actionReqd = len(builderNames) + len(configletNames) != len(applied)`

?

Comment (cheynearista) [reply api/provisioning.go]: I think that would make sense if after line [1810](https://github.com/aristanetworks/go-cvprac/pull/70/files#diff-af964366e44ed2bd45c6078a428dc200R1810) you added a `continue`.  `actionReqd` would need to be set to true if `configletsToRemain` has something different than what `applied` was passed in with (i.e. There were applied configlets removed).

Comment (hansbogert) [reply api/provisioning.go]: I wasn't sure if we partially agreed or not at all, I've redid the logic as per my own suggestion, please review if this logic is sound.

Comment (hansbogert) [reply api/provisioning.go]: @cheynearista gentle nudge :relaxed:

**Comment by cheynearista** [comment]
@hansbogert Sorry...got side tracked on other things. I've merged this into v2 branch and tagged it v2.7.0. My plan is to merge into v3 and develop next.

**Comment by hansbogert** [comment]
> @hansbogert Sorry...got side tracked on other things. I've merged this into v2 branch and tagged it v2.7.0. My plan is to merge into v3 and develop next.

@cheynearista I hope merging into v3 and/or development is still on the radar 8)

**Pull request merged**